src/Zynga/Framework/IO/Disk/V1/Manager.hh

```diff
@@ -0,0 +1,243 @@
+<?hh // strict
+
+namespace Zynga\Framework\IO\Disk\V1;
+
+use \Exception;
+use Zynga\Framework\IO\Disk\V1\Exception\FailedToCloseFileException;
+use Zynga\Framework\IO\Disk\V1\Exception\FailedToCreateDirectoryException;
+use Zynga\Framework\IO\Disk\V1\Exception\FailedToOpenFileException;
+use Zynga\Framework\IO\Disk\V1\Exception\FailedToWriteToFileException;
+use Zynga\Framework\IO\Disk\V1\Exception\ReadPermissionsException;
+use Zynga\Framework\IO\Disk\V1\Exception\WritePermissionsException;
+use Zynga\Framework\IO\Disk\V1\ManagerInterface as DiskIOManagerInterface;
+
+/**
+ * Lightweight class for managing Disk IO
+ */
+class Manager implements DiskIOManagerInterface {
+
+  private static ?DiskIOManagerInterface $instance;
+
+  public static function instance(): DiskIOManagerInterface {
+    if (self::$instance === null) {
+      self::$instance = new self();
+    }
+    invariant(self::$instance !== null, '$instance was null');
+
+    return self::$instance;
+  }
+
+  protected function __construct() {
+  }
+
+  /**
+   * See @ManagerInterface
+   */
+  public function checkOrCreatePath(string $path, int $permissions): bool {
+    if (!$this->doesFileExist($path)) {
+      return $this->makeDirectory($path, $permissions, true);
+    }
+
+    return true;
+  }
+
+  /**
+   * See @ManagerInterface
+   */
+  public function deleteFile(string $path): bool {
+    if ($this->doesFileExist($path)) {
+      return $this->unlink($path);
+    }
+
+    return true;
+  }
+
+  /**
+   * See @ManagerInterface
+   */
+  public function writeFile(string $fileName, string $dataToWrite, int $permissions): void {
+    $filePath = $this->directoryName($fileName);
+    if ($this->checkOrCreatePath($filePath, $permissions) === false) {
+      throw new FailedToCreateDirectoryException($filePath);
+    }
+
+    $handle = $this->fileOpen($fileName, 'w');
```

Comment:
Should we have an append flag here? I think it's fine to truncate each time but I'd note it somewhere.